### ISSUE TITLE
PowerSource.isCharging() not possible?

### ISSUE THREAD

**Issue by stefan-reich** [oshi/oshi#678]
I think in the other threads you said the "is charging" flag was not feasible... is that still true?

I need my AI to remind me to unplug my power source when recording video (microphone buzzes horribly otherwise) :)

**Comment by dbwiddis** [comment]
`PowerSource.getTimeRemaining()` returns -2.0 for "unlimited" which is the result of being charged.  You can check for a value less than -1.5.

**Comment by stefan-reich** [comment]
Yeah I saw this but it doesn't always seem to work... when I unplug for a few seconds, then re-plug, it yields a positive number which is increasing rapidly to absurd heights (like 20 hours remaining or some such).

**Comment by dbwiddis** [comment]
There's lots more we can do with the battery. #306 outlines the options and is a good place to start from.   I'd welcome a PR here.  :)
